doc/neptune-metadata.rst

```diff
@@ -0,0 +1,105 @@
+===========================
+Neptune Metadata Definition
+===========================
+
+This file specifies what fields are in the ``.neptune.yml`` file, what they mean
+and by which tools they are used. The file itself is expected to be in a
+repositories root directory.
+
+.. contents::
+
+library
+-------
+
+Type
+  boolean
+Meaning
+  If set to true, the repository is marked as a library used by other repositories
+Tools
+  - neptune-overview
+  - neptune-autopr
+Example
+  .. code:: yml
+
+    library: true
+
+d2ready
+-------
+
+Type
+  boolean
+Meaning
+  If set to true, the repository is marked as ready for D2 compilation
```

Comment:
Intention was that it may also contain specific strings to distinguish "D1 + D2" from "D2 only" case - it isn't used right now because Makd ignored neptune.yml (CC @leandro-lucarella-sociomantic)

Reply:
So what are the possible values then?

Reply:
Let's settle on these for now:
"true" == "convertable"
"false"
"only"

Reply:
so 
* true
* convertable
* false
* only
with convertable meaning the same as true?

Reply:
Yup

Reply:
What is `true` for? You even mention 3 states in a previous comment:
* D1 only (false)
* D1 + D2 (true)
* D2 only (only)

Maybe it should be more clear, like: `d2-suppport: (no|convertible|d2-only)` or something like that.

Reply:
> What is true for?

Sort of backwards compatibility - it is what is currently used in all projects. For example https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/.neptune.yml#L2